frmMain.cs

```diff
@@ -32,19 +32,24 @@ public frmMain()
 			InitializeComponent();
 		}
 
+		private string GetFilenameWithSuffix(string file, string suffix = "-encrypted")
+		{
+			return System.IO.Path.GetDirectoryName(file) + "\\" + System.IO.Path.GetFileNameWithoutExtension(file) + suffix + ".pdf";
+		}
+
 		private void btnInputBrowse_Click(object sender, EventArgs e)
 		{
+			var previouslySelectedInputFile = txtInputFile.Text;
+
 			DialogResult result = dlgOpen.ShowDialog();
 			if (result == DialogResult.Cancel) { return; }
 
 			txtInputFile.Text = dlgOpen.FileName;
 
-			// Generate output filename matching input with "-encrypted" if none is specified.
-			if (txtOutputFile.Text == "")
+			// Generate output filename matching input with "-encrypted" if none is specified or if overwriting the one generated by default.
+			if (txtOutputFile.Text == "" || txtOutputFile.Text == GetFilenameWithSuffix(previouslySelectedInputFile))
```

Comment:
```suggestion
			if (String.IsNullOrEmpty(txtOutputFile.Text) || txtOutputFile.Text == GetFilenameWithSuffix(previouslySelectedInputFile))
```

Reply:
I've gotten rid of the check completely now 👍